### PR TITLE
OTLP TraceExporter to not populate deprecated fields

### PR DESCRIPTION
**Pull request by cijothomas**
To make https://github.com/open-telemetry/opentelemetry-dotnet/pull/3100/files simpler.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs [hunk 1]
```diff
@@ -252,18 +252,6 @@ private static OtlpTrace.Status ToOtlpStatus(ref TagEnumerationState otlpTags)
                 Code = (OtlpTrace.Status.Types.StatusCode)(int)status,
             };
 
-            if (otlpStatus.Code != OtlpTrace.Status.Types.StatusCode.Error)
-            {
-#pragma warning disable CS0612 // Type or member is obsolete
-                otlpStatus.DeprecatedCode = OtlpTrace.Status.Types.DeprecatedStatusCode.Ok;
```
Comment (alanwest) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs:258]: We could update the protos - this field was removed in 0.12.0. Our protos are 0.11.0. Though, I’d be happy approving this PR and I could follow up with that if you’d prefer.

**Review by alanwest — APPROVED** [review]

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs:258]: let me do it as next PR.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,8 @@
   after instantiation by altering the original `OtlpExporterOptions` provided.
   ([#3066](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3066))
 
+* TraceExporter to stop populating `DeprecatedCode` in OTLP Status.
+
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:21]:
```suggestion
* TraceExporter to stop populating `DeprecatedCode` in OTLP Status.
  ([#3103](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3103))
```

**Review by reyang — APPROVED** [review]
LGTM.

**Pull request merged**